server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceReconciler.java

```diff
@@ -74,23 +77,50 @@ public class DesiredBalanceReconciler {
     /**
      * Number of unassigned shards during last reconciliation
      */
-    protected final AtomicLong unassignedShards = new AtomicLong();
+    protected final LongGaugeMetric unassignedShards;
     /**
      * Total number of assigned shards during last reconciliation
      */
-    protected final AtomicLong totalAllocations = new AtomicLong();
+    protected final LongGaugeMetric totalAllocations;
     /**
      * Number of assigned shards during last reconciliation that are not allocated on desired node and need to be moved
      */
-    protected final AtomicLong undesiredAllocations = new AtomicLong();
+    protected final LongGaugeMetric undesiredAllocations;
+    private final DoubleGauge undesiredAllocationsFraction;
 
-    public DesiredBalanceReconciler(ClusterSettings clusterSettings, ThreadPool threadPool) {
+    public DesiredBalanceReconciler(ClusterSettings clusterSettings, ThreadPool threadPool, MeterRegistry meterRegistry) {
         this.undesiredAllocationLogInterval = new FrequencyCappedAction(threadPool);
         clusterSettings.initializeAndWatch(UNDESIRED_ALLOCATIONS_LOG_INTERVAL_SETTING, this.undesiredAllocationLogInterval::setMinInterval);
         clusterSettings.initializeAndWatch(
             UNDESIRED_ALLOCATIONS_LOG_THRESHOLD_SETTING,
             value -> this.undesiredAllocationsLogThreshold = value
         );
+
+        unassignedShards = LongGaugeMetric.create(
+            meterRegistry,
+            "elasticsearch.allocator.desired_balance_reconciliation.unassigned_shards",
```

Comment:
let's use `es.` prefix instead of elasticsearch. some metrics already do this. naming is under discussion, if the conclusion will be to use a different prefix we will change it